docs/spec_draft.md

```diff
@@ -2963,6 +2969,39 @@ elements of `a` are equal to 1, otherwise the behavior is undefined.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.tuple
+
+### Semantics
+Groups a variadic number of tensors, tokens, or tuples in `val` and produces a
+`result` tuple.
+
+### Inputs
+
+| Name  | Type                                                                |
+|-------|---------------------------------------------------------------------|
+| `val` | variadic number of tensors, tokens, or tuples of any supported type |
+
+### Outputs
+
+| Name     | Type                                                                       |
+|----------|----------------------------------------------------------------------------|
+| `result` | a variadic sized tuple of tensors, tokens, or tuples of any supported type |
+
+### Constraints
+
+  * (C1) For all i, `type(val[i])` = `type(result[i])`.
+
+### Examples
+
+```mlir
+// %var0: [1.0, 2.0]
+// %var1: {[3]}
+%result = "stablehlo.tuple"(%var0, %var1) : (tensor<2xf32>, tuple<tensor<i32>>) -> tuple<tensor<2xf32>, tuple<tensor<i32>>>
```

Comment:
This should say "%val0" and "%val1" to be consistent with how the corresponding input is called.